pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -428,9 +429,11 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 			Dependencies: flow.NewTaskIDs(waitUntilKubeAPIServerIsReady, waitUntilControlPlaneExposureReady, waitUntilControlPlaneExposureDeleted, deployInternalDomainDNSRecord, deployGardenerAccess),
 		})
 		_ = g.Add(flow.Task{
-			Name:         "Sync public service account signing keys to Garden cluster",
-			Fn:           botanist.SyncPublicServiceAccountKeys,
-			SkipIf:       o.Shoot.HibernationEnabled || !v1beta1helper.HasManagedIssuer(botanist.Shoot.GetInfo()),
+			Name: "Sync public service account signing keys to Garden cluster",
+			Fn:   botanist.SyncPublicServiceAccountKeys,
+			SkipIf: o.Shoot.HibernationEnabled ||
+				!features.DefaultFeatureGate.Enabled(features.ShootManagedIssuer) ||
+				!v1beta1helper.HasManagedIssuer(botanist.Shoot.GetInfo()),
```

Comment:
Should the synced keys be deleted in case somebody disables the feature gate after it was enabled?